### PR TITLE
CW-23 Updating the Icon Component so that it receives a link

### PR DESCRIPTION
**Pull request by Irizam**

### CODE REVIEW

File: src/components/icon/index.jsx [hunk 1]
```diff
@@ -9,10 +9,25 @@ import './index.css';
...
-	const { iconName, iconColor, iconBackgroundColor, iconSize, iconBordered } = props;
-	return (
-        <SemanticIcon className={"icon " + iconColor + " " + iconBackgroundColor} name={iconName} size={iconSize} bordered={iconBordered}/>
-	);
+	const { iconName, iconColor, iconBackgroundColor, iconSize, iconBordered, iconHref } = props;
+	const handleClick = (href) => {
+		if(href.includes("mailto:")) {
+			window.location.href = href;
+		}
+		else {
```
Comment (J95000) [inline src/components/icon/index.jsx:17]: you can avoid this statement

File: src/components/icon/index.jsx [hunk 2]
```diff
@@ -9,10 +9,25 @@ import './index.css';
...
+			window.open(href, '_blank');
+		}
+	}
+	if (iconHref !== undefined) {
+		return (
+			<SemanticIcon className={"icon social-icon " + iconColor + " " + iconBackgroundColor} name={iconName} size={iconSize} bordered={iconBordered} onClick={() => handleClick(iconHref ? iconHref : "")}/>
+		);
+	}
+	else {
```
Comment (J95000) [inline src/components/icon/index.jsx:26]: same here

File: src/components/icon/index.jsx (hunk 1 above)
Comment (Irizam) [reply src/components/icon/index.jsx:17]: Why should I avoid these statements?

Comment (J95000) [reply src/components/icon/index.jsx:17]: sorry, avoid the next statement

**Comment by J95000** [comment]
change the branch name it is a defect not a feature

File: src/components/icon/index.jsx (hunk 2 above)
Comment (Irizam) [reply src/components/icon/index.jsx:26]: This if/else statement is used to differentiate an icon used for social media (i.e. YouTube) from one that is only used as an icon, such as the checkmark seen in CW-22

**Comment by Irizam** [comment]
I am updating these changes in a different branch.

**Pull request closed without merging**